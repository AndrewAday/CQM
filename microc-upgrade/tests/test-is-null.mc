struct foo {
  float f;
}

int main()
{
  fmatrix fm;
  struct foo foo;

  if (is_null(foo)) {
    print_string("is null");
  }

  foo = make(struct foo);

  if (is_null(fm)) {
    print_string("is null");
  }

  fm = init_fmat_const(5.,3,3);
  if (!is_null(fm)) {
    print_string("not null");
  }

  if (!is_null(foo)) {
    print_string("not null");
  }

  return 0;
}
