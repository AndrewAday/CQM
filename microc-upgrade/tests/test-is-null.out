is null
is null
not null
not null
